test/Integration/MiddlewareDispatchTest.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-mvc-middleware for the canonical source repository
+ * @copyright Copyright (c) 2019 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-mvc-middleware/blob/master/LICENSE.md New BSD License
+ */
+
+namespace ZendTest\Mvc\Middleware\Integration;
+
+use Interop\Http\ServerMiddleware\MiddlewareInterface;
+use PHPUnit\Framework\TestCase;
+use Prophecy\Argument;
+use stdClass;
+use Zend\Diactoros\Response;
+use Zend\Mvc\Controller\MiddlewareController as DeprecatedMiddlewareController;
+use Zend\Mvc\Middleware\MiddlewareController;
+use Zend\Mvc\MvcEvent;
+use Zend\Router\Http\Literal;
+use ZendTest\Mvc\Middleware\TestAsset\Middleware;
+
+/**
+ * @group integration
+ * @coversNothing
+ */
+class MiddlewareDispatchTest extends TestCase
+{
+    use ApplicationTrait;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->extraConfig = [
```

Comment:
wouldn't be better to pass it into `setUpApplication` instead of setting property?